### ISSUE TITLE
Sync device type changes

### ISSUE THREAD

**Issue by tpx-mk** [ArnesSI/netbox-inventory#187]
Similar to the sync of the Asset-Tag and Serial please add the possibility to sync the device type.

e.g. I'm creating devices via API call from another system, but as the other system doesn't know device types it creates a "dummy device" with the according asset. Afterwards I want to adjust the device type to the proper one but 1. it's not synced to the asset and 2. it's not even possible to change the asset at all (device types of assigned assets can't be changed).

**Comment by tpx-mk** [comment]
Just a little suggestion to at least allow editing the device type (or module type or inventoryitem type) within its own kind:

models.py:
```
[...]
        # if assigned to device/module/inventoryitem we can't change device_type/...
        if (
            self.instance.device or self.instance.module
            or self.instance.inventoryitem
        ):
#            self.fields['manufacturer'].disabled = True
#            for kind in HardwareKindChoices.values():
#                self.fields[f'{kind}_type'].disabled = True

            self.fields[f'device_type'].disabled = not self.instance.device
            self.fields[f'module_type'].disabled = not self.instance.module
            self.fields[f'inventoryitem_type'].disabled = not self.instance.inventoryitem
[...]
```

**Comment by matejv** [comment]
Yeah, this restriction turned out to be a pain in our operations as well. I'll consider allowing changing asset *_type within it's own kind and syncing the type of any related device or module.

Some forms might need additional filters to be easier to select the right asset on AssetDeviceRessign form for example.

**Comment by matejv** [comment]
fixed by #203
